js_modules/dagit/packages/core/src/runs/StepLogsDialog.tsx

```diff
@@ -15,21 +16,50 @@ import {IRunMetadataDict, RunMetadataProvider} from './RunMetadataProvider';
 import {titleForRun} from './RunUtils';
 import {useComputeLogFileKeyForSelection} from './useComputeLogFileKeyForSelection';
 
+export function useStepLogs({runId, stepKeys}: {runId?: string; stepKeys?: string[]}) {
+  const [showingLogs, setShowingLogs] = React.useState<{runId: string; stepKeys: string[]} | null>(
+    null,
+  );
+
+  // Note: This hook captures the runId + stepKeys in state when you click the button and then passes
+  // those values to the modal. This ensures that the modal is "stable" while it's open, even if
+  // the runId passed to the hook changes or becomes undefined. (eg: "Currently materializing" banner)
+
+  return {
+    dialog: (
```

Comment:
this is a good approach I tend to use often. Having a hook that returns components gives the caller more flexibility in terms of where to put things 👍🏾